### PR TITLE
Fum 3196 too big plan

### PR DESCRIPTION
**Pull request by Davidoutz**
This is quite a nasty error that has been reported for quite a long time on different github issues. The issue is that there is that the github runners variables such as GITHUB_STEP_SUMMARY has a max size of whatever shell/bash limitations. As example, the error 

`{{An error occurred trying to start process '/usr/bin/bash' with working directory '/home/runner/work/fum-3137/fum-3137'. Argument list too long}}`

can be reproduced with any simple bash command such as tail, with a too long plan as an argument.

The solution I came up with was to trim the SUMMARY to max 120’000 chars (I think it’s about the right threshold). This way it also works for verrrrrry long plans. I tested with different values from 20’000 up to 100’000 it seems too low, for 150’000 it crashes as well with the same error, therefore I picked arbitrarily 120'000 chars which looks to be fine. The plan I tested on was about 180’000 chars long. This has fixed the issue as you can see here https://github.com/DND-IT/fum-3196/actions/runs/9268960731

Now we can still ask ourselves wether this is an useful step when the plan is too big, because you don't see everything in the output anyway so it's not fully accurate... Maybe we should just return something like "Your plan is too big, check the action" or so instead of cutting it ?  Anyway the logic will be the same

### CODE REVIEW

File: .github/workflows/tf-plan.yaml [hunk 1]
```diff
@@ -115,20 +115,29 @@ jobs:
...
           TERRAFORM_PLAN_PATH: ${{ steps.plan.outputs.text_plan_path }}
         run: |
+          MAX_PLAN_LENGTH=120000
           TERRAFORM_PLAN=$(cat $TERRAFORM_PLAN_PATH)
           TERRAFORM_SUMMARY_HEADER=$(grep -E -i -m 1 'No changes|Plan:|Outputs:' <<< "$TERRAFORM_PLAN")
-
+          PLAN_LENGTH=${#TERRAFORM_PLAN}         
+          START=$((PLAN_LENGTH - MAX_PLAN_LENGTH))
+          if [ "$START" -lt 0 ]; then
+            echo "Plan will not be cut !"
```
Comment (sgametrio) [inline .github/workflows/tf-plan.yaml:124]: @Davidoutz [cosmetic] In the positive scenario, I'd prefer having no extra output (i.e. no `echo` statement)

**Review by swibrow — COMMENTED** [review]
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#step-isolation-and-limits

limit is 1MB

Why not move the change you did to the next job where the issue is then the plans going to the output is fine.

File: .github/workflows/tf-plan.yaml [hunk 2]
```diff
@@ -115,20 +115,29 @@ jobs:
...
+            SHORT_PLAN=${TERRAFORM_PLAN}
+          else
+            echo "Plan will be cut to max ${MAX_PLAN_LENGTH} chars ! (LENGTH: ${PLAN_LENGTH})"
```
Comment (sgametrio) [inline .github/workflows/tf-plan.yaml:127]:
```suggestion
            echo ":warning: Plan output is too long to be displayed (${PLAN_LENGTH} chars). Only the last ${MAX_PLAN_LENGTH} characters will be printed. "
```

**Review by sgametrio — APPROVED** [review]
👍

File: .github/workflows/tf-plan.yaml [hunk 3]
```diff
@@ -115,20 +115,29 @@ jobs:
...
           echo "<details><summary>Click to expand</summary>" >> $GITHUB_OUTPUT
           echo "" >> $GITHUB_OUTPUT
           echo '```terraform' >> $GITHUB_OUTPUT
-          echo "$TERRAFORM_PLAN" >> $GITHUB_OUTPUT
+          echo "${SHORT_PLAN}" >> $GITHUB_OUTPUT
           echo '```' >> $GITHUB_OUTPUT
           echo "</details>" >> $GITHUB_OUTPUT
           echo "${delimiter}" >> $GITHUB_OUTPUT
-
+
```
Comment (rbapst-tamedia) [inline .github/workflows/tf-plan.yaml:140]:
```suggestion

```

**Review by rbapst-tamedia — APPROVED** [review]

**Pull request merged**